app/models/user.rb

```diff
@@ -3,9 +3,14 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy
   has_many :posts, foreign_key: 'author_id', dependent: :destroy
 
-  validates :name, presence:
+  validates :name, presence: true
+  validates :posts_counter, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def recent_posts(limit = 3)
     posts.order(created_at: :desc).limit(limit)
   end
+
+  def update_user_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
Since it is a requirement that you add unit specs for all of your models' methods and validations. kindly add tests for your ` update_user_posts_counter` and  `recent_posts` method in the user.spec.rb😉.

<img width="736" alt="image" src="https://github.com/Omwomo/eugene-mindscape/assets/64643995/908d7b86-8c87-43dd-b60a-242e562deff5">